### PR TITLE
Closes #5041 on rucss compatibility with inline related posts dynamic selectors

### LINKED ISSUE #5041: RUCSS - Inline Related Posts widget breaks due to dynamic inline selector
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version - yes
- Used the search feature to ensure that the bug hasn’t been reported before - yes

**Describe the bug**
The Inline Related Posts plugin adds inline style with a selector like this `.u7cb7434f55ddbc61829903e3a7b5fb85`. 
The numeric-letter value is dynamic and the layout breaks when cache is refreshed, but Used CSS is still referencing an old selector.

There is unfortunately no unique `content` we can trigger. The style tag doesn't have any `id` or `attribute`.

The inline style looks like this:
image
Screenshot: https://i.imgur.com/EtuGf2Q.png

`.ctaText` is probably the only `content` we could use, but it is not unique.

**To Reproduce**
1. Install and activate **Inline Related Posts** plugin;
2. Save settings for the plugin - can be used default;
3. Create a post and add Related Posts module with a block;
4. Enable Remove Unused CSS option;
5. Used CSS will include the initial dynamic selector.
6. After cache is cleared the selector will change and styling will be missing.

**Expected behavior**
The style for the Related Posts module needs to be preserved.

**Additional context**
Plugin: Inline Related Posts https://wordpress.org/plugins/intelly-related-posts/
80 000 installations

Ticket: https://secure.helpscout.net/conversation/1875506582/341695/

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by jeawhanlee**
## Description

Preserve dynamic inline styles for inline related posts plugin.

Fixes #5041 

## Type of change

- [x] Enhancement (non-breaking change which improves an existing functionality)

## Is the solution different from the one proposed during the grooming?

This solution is in no way different from the proposed.

## How Has This Been Tested?

- Install & Activate inline related posts.
- Activate RUCSS
- Created Identical Posts.
- Then inspect page to see that dynamic inline styles are preserved

# Checklist:

Please delete the options that are not relevant.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: inc/ThirdParty/Plugins/InlineRelatedPosts.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		if ( ! defined( 'IRP_PLUGIN_SLUG' ) ) {
+			return [];
+		}
+
+		$events['rocket_rucss_inline_content_exclusions'] = 'exclude_inline_from_rucss';
+
+		return $events;
```
Comment (remyperona) [inline inc/ThirdParty/Plugins/InlineRelatedPosts.php]:
```suggestion
		return [
		    'rocket_rucss_inline_content_exclusions' => 'exclude_inline_from_rucss',
		];
```

File: tests/Integration/inc/ThirdParty/Plugins/InlineRelatedPosts/excludeInlineFromRucss.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * @covers  \WP_Rocket\ThirdParty\Plugins\InlineRelatedPosts::exclude_inline_from_rucss
+ * @group   InlineRelatedPosts
+ */
+class Test_ExcludeInlineFromRucss extends TestCase {
+
+	public function tear_down() {
+		remove_filter( 'rocket_rucss_inline_content_exclusions', [ $this, 'return_false' ] );
+
+		parent::tear_down();
+	}
```
Comment (remyperona) [inline tests/Integration/inc/ThirdParty/Plugins/InlineRelatedPosts/excludeInlineFromRucss.php]: This is not necessary

File: tests/Integration/inc/ThirdParty/Plugins/InlineRelatedPosts/excludeInlineFromRucss.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+	/**
+	 * @dataProvider configTestData
+	 */
+	public function testShouldReturnExpected( $config, $expected ) {
+        $inline_related_posts = new InlineRelatedPosts();
+		$this->assertSame( $expected, apply_filters( 'rocket_rucss_inline_content_exclusions', $inline_related_posts->exclude_inline_from_rucss($config['excluded'] ) ) );
```
Comment (remyperona) [inline tests/Integration/inc/ThirdParty/Plugins/InlineRelatedPosts/excludeInlineFromRucss.php]: You need to either use the filter, or the method call, but not both at the same time in the integration test.

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by remyperona — APPROVED** [review]

**Review by Mai-Saad — APPROVED** [review]
Working as expected

**Pull request merged**